base/inference.jl

```diff
@@ -63,6 +63,13 @@ immutable Const
     Const(v::ANY) = new(v)
 end
 
+immutable PartialTypeVar
+    tv::TypeVar
+    lb_certain::Bool
+    ub_certain::Bool
```

Comment:
I would define this as

```
immutable PartialTypeVar
    name
    lb
    ub
end
```

where the components are simply the inferred types of the arguments to `TypeVar`. You know they're certain if they're `Const`. Did you come across any reasons this approach wouldn't work?